guides/source/caching_with_rails.md

```diff
@@ -295,14 +295,14 @@ Consider the following example. An application has a `Product` model with an ins
 ```ruby
 class Product < ApplicationRecord
   def competing_price
-    Rails.cache.fetch("#{cache_key}/competing_price", expires_in: 12.hours) do
+    Rails.cache.fetch("#{cache_key_with_version}/competing_price", expires_in: 12.hours) do
       Competitor::API.find_price(id)
     end
   end
 end
 ```
 
-NOTE: Notice that in this example we used the `cache_key` method, so the resulting cache key will be something like `products/233-20140225082222765838000/competing_price`. `cache_key` generates a string based on the model's `id` and `updated_at` attributes. This is a common convention and has the benefit of invalidating the cache whenever the product is updated. In general, when you use low-level caching for instance level information, you need to generate a cache key.
+NOTE: Notice that in this example we used the `cache_key_with_version` method, so the resulting cache key will be something like `products/233-20140225082222765838000/competing_price`. This method generates a string based on the model's `id` and `updated_at` attributes. There is also a `cache_key` method that does not include a timestamp unless you turn off `ActiveRecord::Base.cache_versioning`. The timestamp has the benefit of invalidating the cache whenever the product is updated. In general, when you use low-level caching for instance level information, you need to generate a cache key.
```

Comment:
According to [the documentation on `ActiveRecord::Base.cache_versioning`](https://github.com/rails/rails/blob/b5302d5a820b078b6488104dd695a679e5a49623/activerecord/lib/active_record/integration.rb#L23), it is going to be true by default on 6.0's release. Do we really need to change these docs on master in that case? Seems like maybe we should just backport a find replace of `cache_key` => `cache_key_with_version` to the guides for `5.2`.

Reply:
@gmcgibbon
Thanks!
I returned the note to the original and just replaced `cache_key` with `cache_key_with_version`.
Since it was also used in the "Conditional GET support" section, I replaced it.

Reply:
Perfect! Please squash your commits and I'll merge. Any concerns with this @rafaelfranca?

Reply:
Actually, one last thing. Can you also change your base branch to `5-2-stable`. Master has Rails 6 on it currently but this is specific to 5.2.

Reply:
@gmcgibbon 

I cleaned my commits!

`cache_versioning` will be `true` by default on Rails 6.0 and [it be also true on Rails 5.2](https://github.com/rails/rails/blob/v5.2.1/railties/lib/rails/application/configuration.rb#L94).
`cache_key` returns key without a timestamp if `cache_versioning` is `true`. 
I think that this change is also necessary for master. What do you think?

```
Product.cache_versioning = false     # 5.1 or earlier default
Product.last.cache_key               # => "products/1-20181116023527886197"
Product.last.cache_key_with_version  # => "products/1-20181116023527886197"

Product.cache_versioning = true      # 5.2 or later default
Product.last.cache_key               # => "products/1"
Product.last.cache_key_with_version  # => "products/1-20181116023527886197"
```

Reply:
You are correct. 👍 Sorry, that comment confused me about the current behaviour.